lib/Coffer/Path.hs

```diff
@@ -221,16 +224,31 @@ instance (Buildable path) => Buildable (QualifiedPath path) where
       Nothing -> build path
 
 instance (FromHttpApiData path) => FromHttpApiData (QualifiedPath path) where
-  parseUrlPiece text
-    | [pathPiece] <- T.splitOn "#" text = do
-        path <- parseUrlPiece @path pathPiece
-        pure $ QualifiedPath Nothing path
-    | [backendPiece, pathPiece] <- T.splitOn "#" text = do
-        backendName <- newBackendName backendPiece
-        path <- parseUrlPiece @path pathPiece
+  parseUrlPiece = mkQualifiedPath
+    (\_ -> "Invalid qualified path format. Expected [BACKENDNAME#]PATH")
+    parseUrlPiece
+
+mkQualifiedPath
+  :: (Text -> Text)
+  -> (Text -> Either Text path)
+  -> (Text -> Either Text (QualifiedPath path))
+mkQualifiedPath errorMessage mkPath text = case T.splitOn "#" text of
+      [backendNameStr, pathStr] -> do
+        backendName <- readBackendName' backendNameStr
```

Comment:
We can just use `newBackendName` directly

```suggestion
        backendName <- newBackendName backendNameStr
```

Reply:
Done